File: 7term/web/lab4/server/src/entities/ComputerSpecs.ts
import { Entity, PrimaryGeneratedColumn, Column, Check } from "typeorm";

@Entity()
@Check(`"ram" > 0`)
@Check(`"storage" > 0`)
export class ComputerSpecs {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  cpu: string;

  @Column()
  ram: number;

  @Column()
  storage: number;
}

File: 7term/web/lab6/server/src/app/models/computer_specs.py
from sqlalchemy import Column, Integer, String, CheckConstraint

from ..core.db.database import Base


class ComputerSpecs(Base):
    __tablename__ = "computer_specs"

    id = Column(Integer, primary_key=True, index=True)
    cpu = Column(String, nullable=False)
    ram = Column(Integer, nullable=False)
    storage = Column(Integer, nullable=False)

    __table_args__ = (
        CheckConstraint("ram > 0", name="check_ram_positive"),
        CheckConstraint("storage > 0", name="check_storage_positive"),
    )
